server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/controllers/ToolController.java

```diff
@@ -0,0 +1,197 @@
+import edu.illinois.ncsa.incore.common.auth.Authorizer;
+import edu.illinois.ncsa.incore.common.auth.IAuthorizer;
+import edu.illinois.ncsa.incore.common.dao.ISpaceRepository;
+import edu.illinois.ncsa.incore.common.exceptions.IncoreHTTPException;
+import edu.illinois.ncsa.incore.common.utils.UserGroupUtils;
+import edu.illinois.ncsa.incore.common.utils.UserInfoUtils;
+import edu.illinois.ncsa.incore.service.project.dao.IProjectRepository;
+import edu.illinois.ncsa.incore.service.project.models.*;
+import io.swagger.v3.oas.annotations.OpenAPIDefinition;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.info.Contact;
+import io.swagger.v3.oas.annotations.info.Info;
+import io.swagger.v3.oas.annotations.info.License;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.*;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.apache.log4j.Logger;
+import org.glassfish.jersey.media.multipart.FormDataMultiPart;
+import org.glassfish.jersey.media.multipart.MultiPartFeature;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@OpenAPIDefinition(
+    info = @Info(
+        description = "IN-CORE Project Service for creating and accessing tools",
+        version = "1.28.0",
+        title = "IN-CORE v2 Project Service API",
+        contact = @Contact(
+            name = "IN-CORE Dev Team",
+            email = "incore-dev@lists.illinois.edu",
+            url = "https://tools.in-core.org"
+        ),
+        license = @License(
+            name = "Mozilla Public License 2.0 (MPL 2.0)",
+            url = "https://www.mozilla.org/en-US/MPL/2.0/"
+        )
+    )
+)
+// Not sure if @Tag is equivalent to @Apis
+@Tag(name = "tools")
+
+@Path("tools")
+public class ToolController {
+    private final Logger logger = Logger.getLogger(ToolController.class);
+
+    private final String username;
+    private final List<String> groups;
+    private final String userGroups;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Inject
+    private IProjectRepository projectDAO;
+
+    @Inject
+    private ISpaceRepository spaceRepository;
+
+    @Inject
+    private IAuthorizer authorizer;
+
+    @Inject
+    public ToolController(
+        @Parameter(name = "User credentials.", required = true) @HeaderParam("x-auth-userinfo") String userInfo,
+        @Parameter(name = "User groups.", required = false) @HeaderParam("x-auth-usergroup") String userGroups
+    ) {
+        this.userGroups = userGroups;
+        this.username = UserInfoUtils.getUsername(userInfo);
+        this.groups = UserGroupUtils.getUserGroups(userGroups);
+    }
+
+    @POST
+    @Path("/bld-inventory")
```

Comment:
shouldn't this have {projectid} here? I am not sure ... so the endpoint becomes `project/api/tools/{projectid}/bld-inventory` ?

Reply:
or `project/api/{projectid}/tools/bld-inventory`